agent/consul/rpc.go

```diff
@@ -521,6 +522,23 @@ func (s *Server) blockingQuery(queryOpts structs.QueryOptionsCompat, queryMeta s
 	var timeout *time.Timer
 	var queryTimeout time.Duration
 
+	// REVIEW: The spec for this placed the atomic inc for queries_blocking inside the RUN_QUERY statement
+	// however, there are a combination of factors that I believe should mean we should place it before.
+	// - `goto RUN_QUERY` on L545 above means that we'd instrument a non-blocking query as blocking for its duration.
```

Comment:
Doesn't placing it here is also going guarantee we change the blocking gauge for all (even non-blocking) RPCs?Was you intent here to explain that it's _no worse_ than inside if it's here? I read this as "it needs to be here _otherwise_ we count them all.

Overall I 100% agree it's better here to keep defer semantics sane but I'd suggest one of two things, either:
 1. Still gate this on `queryOpts.GetMinQueryIndex() > 0` that way it only counts queries that actually _intend_ to block which is the semantics the issue suggested.
   * other benefit is less churn on that atomic counter and metrics gauge for every single read RPC as well as IMO more a more useful metric - showing the number of active blocking calls vs total number of in-flight read RPCs.
 2. If you think including non-blocking reads is valuable to include here, I suggest we'd at least need to change the metric name as "queries_blocking" implies that only blocking queries are being counted.

My vote would be #1 I think and it's super simple but open to other semantics if you think they are useful?

Reply:
Thank you for catching this bug. I definitely intended the semantics in 1 and had that set up in a prior commit in this PR. I thoughtlessly moved it above while fixing a syntax error w/ declaring variables after a goto jump. That's addressed easily enough now by _actually_ declaring at the top of the fn, rather than moving the whole set of operations. 

Should be fixed in latest revision.